mongo/readconcern/readconcern_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package readconcern_test
+
+import (
+	"testing"
+
+	"go.mongodb.org/mongo-driver/internal/assert"
+	"go.mongodb.org/mongo-driver/mongo/readconcern"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+func TestReadConcern_MarshalBSONValue(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name      string
+		rc        *readconcern.ReadConcern
+		bytes     []byte
+		wantError error
+	}{
+		{
+			name:      "local",
+			rc:        &readconcern.ReadConcern{Level: "local"},
```

Comment:
Optional: use the "Local" constructor:

```suggestion
			rc:        readconcern.Local(),
```